core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -247,26 +247,25 @@ protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
   @Override
   public boolean dropTable(TableIdentifier identifier, boolean purge) {
     if (!isValidIdentifier(identifier)) {
-      throw new NoSuchTableException("Invalid identifier: %s", identifier);
```

Comment:
This is actually a dead code block for this class. If you check isValidIdentifier it is ```return true```. I think we should just leave this bit as is for now and maybe clean it up in a future pr.

Reply:
sure, have rolled back my changes for this section.